transport/http/inbound.go

```diff
@@ -71,6 +72,20 @@ func (h httpHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	defer req.Body.Close()
 
+	caller := req.Header.Get(CallerHeader)
+	if len(caller) == 0 {
+		http.Error(w, "caller name is required", http.StatusBadRequest)
+		return
+	}
+	req.Header.Del(CallerHeader)
+
+	service := req.Header.Get(ServiceHeader)
+	if len(service) == 0 {
+		http.Error(w, "service name is required", http.StatusBadRequest)
+		return
+	}
```

Comment:
Extracting this information from the `req` should probably be a function at this point.